central/telemetry/centralclient/interceptors.go

```diff
@@ -0,0 +1,107 @@
+package centralclient
+
+import (
+	"strings"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/telemetry/phonehome"
+)
+
+var (
+	trackedPaths set.FrozenSet[string]
+	ignoredPaths = []string{"/v1/ping", "/v1.PingService/Ping", "/v1/metadata", "/static/"}
+
+	uninitializedClusters = set.NewSet[string]()
+	mux                   = &sync.Mutex{}
+
+	interceptors = map[string][]phonehome.Interceptor{
+		"API Call":            {apiCall},
+		"Cluster Registered":  {clusterRegistered},
+		"Cluster Initialized": {clusterInitialized},
+		"roxctl":              {roxctl},
+	}
+)
+
+// Adds Path, Code, User-Agent and Method properties to API Call events for the
+// API paths specified in the ROX_TELEMETRY_API_WHITELIST central deployment
```

Comment:
As someone not familiar with the codebase, I don't understand how `ROX_TELEMETRY_API_WHITELIST` is related to this function. One should bring context from different place to understand that `ROX_TELEMETRY_API_WHITELIST` impacts `apiWhiteList` which in turn impacts `trackedPaths` which is used here (but initialized in a different file).